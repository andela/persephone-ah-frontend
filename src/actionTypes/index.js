export const SAMPLE_CONSTANT = 'SAMPLE_CONSTANT';
export const APP_THEME = 'APP_THEME';
