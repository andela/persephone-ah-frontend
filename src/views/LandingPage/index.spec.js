import React from 'react';
import { shallow, mount } from 'enzyme';
import * as actionTypes from '../../actionTypes';
import { themeToggler } from './index.action';
import { LandingPage } from './index';

const defaultProps = {
  themeToggler: jest.fn(),
  theme: {}
};

const actionArgs = {
  theme: 'light-theme'
};

describe('Render component', () => {
  it('should create an action ', () => {
    //arrange
    const expectedAction = {
      type: actionTypes.APP_THEME,
      payload: actionArgs
    };
    // act
    const action = themeToggler(actionArgs);
    //assert
    expect(action).toEqual(expectedAction);
  });

  it('should render component successfully', () => {
    const component = shallow(<LandingPage {...defaultProps} />);
    expect(
      component.contains(<h1>Team Persephone - Author's Haven</h1>)
    ).toBeTruthy();
  });

  it('should toggle to light theme', () => {
    const component = mount(<LandingPage {...defaultProps} />);
    component.find('button').simulate('click');
    expect(component.find('button').hasClass('')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('should toggle to dark theme', () => {
    const component = mount(<LandingPage {...defaultProps} />);
    component.find('button').simulate('click');
    expect(component.find('button').hasClass('switch')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
