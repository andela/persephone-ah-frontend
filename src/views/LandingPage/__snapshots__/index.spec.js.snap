// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component should toggle to dark theme 1`] = `ReactWrapper {}`;

exports[`Render component should toggle to light theme 1`] = `ReactWrapper {}`;
